tools/refguide_check.py

```diff
@@ -810,6 +859,50 @@ def check_doctests_testfile(fname, verbose, ns=None,
     return results
 
 
+def iter_included_files(base_path, verbose=0, suffixes=('.rst',)):
+    if os.path.exists(base_path) and os.path.isfile(base_path):
+        yield base_path
+    for dir_name, subdirs, files in os.walk(base_path, topdown=True):
+        if dir_name in RST_SKIPLIST:
+            if verbose > 0:
+                sys.stderr.write('skipping files in %s' % dir_name)
+            files = []
+        for p in RST_SKIPLIST:
+            if p in subdirs:
+                if verbose > 0:
+                    sys.stderr.write('skipping %s and subdirs' % p)
+                subdirs.remove(p)
+        for f in files:
+            if (os.path.splitext(f)[1] in suffixes and
+                    f not in RST_SKIPLIST):
+                yield os.path.join(dir_name, f)
+
+
+def check_documentation(base_path, results, args, dots):
+    """Check examples in any *.rst located inside `base_path`.
+    Add the output to `results`.
+
+    See Also
+    --------
+    check_doctests_testfile
+    """
+    for filename in iter_included_files(base_path, args.verbose):
+        if dots:
+            sys.stderr.write(filename + ' ')
+            sys.stderr.flush()
+
+        tut_results = check_doctests_testfile(filename,
+            (args.verbose >= 2), dots=dots,
+            doctest_warnings=args.doctest_warnings)
+
+        def scratch(): pass        # stub out a "module", see below
```

Comment:
Should be a blank line after function def.  Also, put `pass` on separate line.

Reply:
adopted